### ISSUE TITLE
Generate training dataset for Layer 3

### ISSUE THREAD

**Issue by MustafaKhan0** [alansun17904/MAAA#25]
# Summary

Generate the dataset used for training the model. Specifically, the training dataset is composed of the factual inputs, which should be passed to the model, and the counter-factual outputs. The counter-factual outputs are the outputs of the model when it is passed a counter-factual input. 

Branch should be created from the layer3 branch. 

Description from the research proposal:

Edge pruning requires storing the edge mask $z$ whose size scales with the actual size of the model. Herein, we propose a modification to the core procedure of edge pruning that does not require edge masks. We first construct a counterfacutal dataset: for each input $x$ and counterfactual input $x'$ we yield model predictions $M(x), M(x')$. Then, we finetune $M$ to create $M^\star$. 

# Proposed Changes

Create a function / multiple function to achieve this generation of the dataset. This should then be imported and called in src/layer3/test.ipynb (for now). 

Later, we will change the Jupyter notebook to a python file. 

# Acceptance Criteria

The `training_dataset` variable in src/layer3/test.ipynb should have the correct training info, ready to be used by an HF / HF Compatible Trainer.

**Comment by MustafaKhan0** [comment]
Made the changes on the layer3 branch - will discuss and review during the next meeting, and test after #26 is complete!
